src/Education/SonarRuleIdUriEncoderDecoder.cs

```diff
@@ -36,5 +38,28 @@ public static Uri EncodeToUri(SonarCompositeRuleId compositeRuleId)
 
             return new Uri(link);
         }
+
+        public static bool TryDecodeToCompositeRuleId(Uri uri, out SonarCompositeRuleId compositeRuleId)
+        {
+            compositeRuleId = null;
+
+            // If the URI can be decoded it would be in the form of 'sonarlintrulecrossref://ruleRepo/ruleKey'
+            if (uri.Scheme != prefix)
+            {
+                return false;
+            }
+
+            // It is not required to use uri.Host here. It would have been fine to use any
+            // other property that had the information on the repository. 'host' is just the most simple.
+            var ruleRepo = uri.Host;
+
+            Debug.Assert(uri.AbsolutePath[0] == '/',
```

Comment:
We really aren't ever expecting this to happen however, I am not sure what to do if it does? As of right now it would just crash but I dont know until what point I can assume to get the correct input.

Reply:
It's fine if we crash on bad input here.

The assertion has two purposes: it documents an assumption we are making about how this method is expected to be called, and acts as a notification in debug builds if that assumption is not met (i.e. highlights there is a problem and makes it easy to find). It's not intended to prevent a crash.